### ISSUE TITLE
Trackers Not Blocked

### ISSUE THREAD

**Issue by Nzzyme** [MuntashirAkon/AppManager#1058]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

Even though App Manager is used to block trackers from an app, they are not actually blocked as observed through the filtering log of AdGuard (which shows connections still being made by the app to the tracking/ad servers).

### To Reproduce

I have installed AM (v3.2.0-alpha01) on my rooted Android TV Box running Android 8.1.0. I also have AdGuard installed to monitor the connections made by apps

Example App Used for Testing: https://play.google.com/store/apps/details?id=com.audioaddict.zr

Steps:

1. Open AM app. All apps are listed.
2. Click on an app (which has trackers).
3. On the next page (app details page), we see the tracker count. Click on it.
4. The tracker window opens with all the trackers listed. Click 'Select All' and then on 'Block'.
5. On the same page (app details page), click on the 'Launch' button to open the app
6. Switch to the filtering log of AdGuard app and I still see connections being made by the target app to those trackers which I had blocked in AM app.

Note: The app used for testing (Zen Radio) was excluded from Ad blocking in AdGuard and DNS filters were turned off.

### Expected behavior

If the trackers were blocked by AM, then ideally there should not be any calls being made by the app to the tracking/ad server.

### Device info

- Device: A95X Max
- OS Version: Android 8.1.0
- App Manager Version: v3.2.0-alpha01
- Mode: Root

**Comment by MuntashirAkon** [comment]
I think you're confusing trackers with tracker components. Read [the docs](https://muntashirakon.github.io/AppManager/en/#sec:terminologies) to understand the terminologies used in App Manager. In this particular case, it is the following paragraph:

> **Tracker** — Denotes tracker components throughout the document and in App Manager except in the [scanner page](https://muntashirakon.github.io/AppManager/en/#sec:scanner-page). Trackers include libraries such as crash reporters, analytics, profiling, identification, ad, location, etc. Thus, they are not equal in functions. There is no distinction or bias between open source and closed source libraries that promote tracking.

TL;DR; App Manager is NOT a network firewall. Even if you use App Manager to block tracker components, you'd still need a firewall app such as pDNSf. Now, you might ask what's the benefit of blocking tracker components and why doesn't it prevent the app from calling _home_.

Tracker components are used to collect information (e.g. services, broadcast receivers), display advertisements (e.g. activities, services) or share information (e.g. providers). As a result, blocking these components would prevent those operations, but they do not necessarily stop calling home as they might be triggered by a different component which hasn't been blocked yet or couldn't be blocked at all. But blocking those components will ensure that even if the app still calls home, it will not send any data because it couldn't collect any. This also improves the performance of the app as well as reduce memory usage since many of these components tend to utilise a lot of memory in order to collect data.

**Comment by Nzzyme** [comment]
Thanks for the explanation. That is very surprising. Well, most people like me are under the same impression that blocking 'trackers' blocks them completely from the target app. So what is being blocked is the ability to collect tracker data.

It may be included in the documentation but frankly, it is obfuscated and not that easy to understand for a common user without Android knowledge :)

So, in short, even after blocking the 'tracker components', the app may still call to those tracking/ad servers but will NOT send / receive any data whatsoever? Am I right in understanding this? In that case, why would I again need another firewall app? I mean let the app connect to the tracking servers but there is nothing to track as the app will not send/receive any tracking data?

**Comment by MuntashirAkon** [comment]
> That is very surprising. Well, most people like me are under the same impression that blocking 'trackers' blocks them completely from the target app. So what is being blocked is the ability to collect tracker data.

Yes. This is why this name will be replaced with alternatives #827. They were called trackers for historical reasons (i.e. **ClassyShark3xodus** calls them trackers) which seems to be very misleading. Also, trackers often have a very broad definition and other times a very narrow definition, which too is problematic.

> So, in short, even after blocking the 'tracker components', the app may still call to those tracking/ad servers but will NOT send / receive any data whatsoever? Am I right in understanding this?

Yes, they cannot send or receive any meaningful data.

> In that case, why would I again need another firewall app? I mean let the app connect to the tracking servers but there is nothing to track as the app will not send/receive any tracking data?

This depends on your threat model. If it includes zero communications between offending servers (e.g. Google/Firebase Analytics), you need to use a firewall app to ensure that this never happens. App Manager could, of course, include a network firewall (#80), but since there are many different approaches to network firewall, I didn't find it suitable for App Manager.
